bus/bus.go

```diff
@@ -852,53 +852,91 @@ func (b *Bus) renewContractV2(ctx context.Context, cs consensus.State, h api.Hos
 	if gb := gc.CheckV2(settings); gb.Gouging() {
 		return api.ContractMetadata{}, errors.New(gb.String())
 	}
+	prices := settings.Prices
 
-	// determine the new collateral we want the host to put into the
-	// renewed/refreshed contract
-	newCollateral := settings.Prices.Collateral.Mul64(expectedNewStorage).Mul64(endHeight - cs.Index.Height)
-	if newCollateral.Cmp(minNewCollateral) < 0 {
-		newCollateral = minNewCollateral
+	// target the max collateral the host is willing to accept for the renterFunds
+	collateral := rhpv4.MaxHostCollateral(prices, renterFunds)
+
+	// cap the collateral at the per-contract MaxCollateral of the host
+	totalCollateral := collateral.Add(rev.TotalCollateral)
+	if totalCollateral.Cmp(settings.MaxCollateral) > 0 {
+		collateral = collateral.Sub(totalCollateral.Sub(settings.MaxCollateral))
```

Comment:
I think this can underflow if `rev.TotalCollateral` was already higher than `settings.MaxCollateral`... right? Definitely a dangerous `sub` here imo. Not sure if it's legal but a host can update his settings after forming the contract